crypto/asn1/asn1_lib.c

```diff
@@ -272,13 +272,22 @@ int ASN1_STRING_set(ASN1_STRING *str, const void *_data, int len)
 {
     unsigned char *c;
     const char *data = _data;
+    size_t new_len = 0;
 
     if (len < 0) {
         if (data == NULL)
             return 0;
         else
-            len = strlen(data);
+            new_len = strlen(data);
     }
+
+    if (new_len > INT_MAX) {
+        ASN1err(ASN1_F_ASN1_STRING_SET, ASN1_R_LENGTH_OUT_OF_RANGE);
+        return 0;
+    } else {
+        len = (int)new_len;
+    }
```

Comment:
This block looks wrong. Surely this should all be moved into the `if (len < 0)` block? Otherwise we overwrite `len` even if it is positive when passed to this function.

Reply:
You are right, good catch! If `len > 0` then `new_len` will always be 0, so `len` is reset to 0.

Reply:
This proves that it's not a trivial change ;-)